src/form-builder/components/FormPrivilegeTable.jsx

```diff
@@ -0,0 +1,364 @@
+                    .then((data) => {
+                      const parsedFormValue = data.resources.length > 0 ?
+                        JSON.parse(data.resources[0].value) : {};
+
+                      this.setState({   formData: data,
+                                        httpReceived: true,
+                                        loading: false,
+                                        formName: data.name,
+                                        referenceVersion: parsedFormValue.referenceVersion,
+                                        referenceFormUuid: parsedFormValue.referenceFormUuid
+                                    });
+
+
+                      //this.props.dispatch(formLoad(formControlsArray));
+                    })
+                    .catch((error) => {
+                      this.setErrorMessage(error);
+                      this.setState({ loading: false });
+                    });
+}
+  fetchPrivileges() {
+    let initialPrivileges = [];
+    const queryParams = `?=`;
+    const optionsUrl = `${formBuilderConstants.formPrivilegeUrl}${queryParams}`;
+     httpInterceptor.get(optionsUrl)
+        .then((initialPrivileges) => {
+               this.setState({ availablePrivileges: this.orderFormByVersion(initialPrivileges.results), loading: false });
+        })
+
+}
+ orderFormByVersion(privilege) {
+    const sampleList=[];
+     privilege.forEach((privilege) => {
+       const item ={
+                 value:privilege.display,
+                 uuid: privilege.uuid,
+                 label:privilege.display,
+                 }
+       sampleList.push(item);
+       });
+    return sampleList;
+  }
+
+
+ handleTag(event,idx) {
+  const formPrivileges = this.state.formPrivileges.slice();
+  const availablePrivileges = this.state.availablePrivileges.slice();
+        const tempItem = {
+             uuid: event.uuid,
+            privilegeName: event.display,
+            isEditable: false,
+            isViewable: false,
+          };
+
+       if(event.event != undefined && (formPrivileges.length >0) && (event.event.currentTarget.type === "checkbox")){
+
+            if(event.event.currentTarget.name === 'isEditable'){
+                formPrivileges[event.idx].isEditable = event.event.currentTarget.checked;
+            }
+            if(event.event.currentTarget.name === 'isViewable'){
+                formPrivileges[event.idx].isViewable = event.event.currentTarget.checked;
+            }
+
+       }
+       if(event != undefined && (event.event == undefined)){
+            formPrivileges[idx].privilegeName = event.value;
+            formPrivileges[idx].uuid = event.uuid;
+            this.setState({selectedPrivilegeOption: event.value});
+       }
+          this.setState({formPrivileges:formPrivileges});
+         // this.removeSelectedPrivilege(event.value);
+     };
+ handleAddRow(){
+    var id = (+ new Date() + Math.floor(Math.random() * 999999)).toString(36);
```

Comment:
@buvaneswari-arun do we have any existing math utils so that we can move this code there ?